python/src/main/python/job-builder-server/requirements.txt

```diff
@@ -386,9 +386,9 @@ google-apitools==0.5.31 \
     --hash=sha256:4af0dd6dd4582810690251f0b57a97c1873dadfda54c5bc195844c8907624170 \
     --hash=sha256:6be92c1c3e93485450420bb0e365d47eb4d8a835d03ebe1963dc6da4d39a7b0e
     # via apache-beam
-google-auth==2.39.0 \
-    --hash=sha256:0150b6711e97fb9f52fe599f55648950cc4540015565d8fbb31be2ad6e1548a2 \
-    --hash=sha256:73222d43cdc35a3aeacbfdcaf73142a97839f10de930550d89ebfe1d0a00cde7
+google-auth==2.38.0 \
```

Comment:
It's installing older versions. It appears one needs to delete all `**/__build__` then run the script to get a fresh environment

Reply:
I see. I ran the tool on a fresh environment and get

```
Traceback (most recent call last):
  File "/usr/local/google/home/yathu/DataflowTemplates/__build__/python_requirements_gen/bin/pip-compile", line 8, in <module>
...
  File "/usr/local/google/home/yathu/DataflowTemplates/__build__/python_requirements_gen/lib/python3.11/site-packages/piptools/repositories/pypi.py", line 452, in allow_all_wheels
    self.finder.find_all_candidates.cache_clear()
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'function' object has no attribute 'cache_clear'
```

build requirement wasn't successful and caused old dependency ran on last release validation get copied

Reply:
This is due to the latest pypi 25.1. Downgrade to 25.0.1 in python/generate_dependencies.sh solved the issue.

Reply:
Fixed